e2e/e2e_test.go

```diff
@@ -0,0 +1,2610 @@
+//go:build e2e
+// +build e2e
+
+/*
+Package e2e provides an end-to-end test suite for the Functions CLI "func".
+
+See README.md for more details.
+*/
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"knative.dev/func/cmd"
+	fn "knative.dev/func/pkg/functions"
+	"knative.dev/func/pkg/knative"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/func/pkg/k8s"
+)
+
+const (
+	// DefaultBin is the default binary to run, relative to this test file.
+	// This is the binary built by default when running 'make'.
+	// This can be customized with FUNC_E2E_BIN.
+	// NOte this is always relative to this test file.
+	DefaultBin = "../func"
+
+	// DefaultClean indicates whether or not tests should clean up after
+	// themselves by deleting Function instances created during run.
+	// Setting this to false significantly increases testing speed, but
+	// results in lingering function instances after at test run.  Set to
+	// "false" when expecting the test cluster to be removed after a test run,
+	// such as in CI, but set to "true" for development, when the same test
+	// cluster may be used across multiple test runs during debugging.
+	DefaultClean = true
+
+	// DefaultCleanImages indicates whether or not tests should clean up container
+	// images and volumes after themselves. This is separate from DefaultClean
+	// which handles cluster resources. This is necessary in CI which quickly
+	// runs out of disk space during Quarkus and Springboot builds.
+	// Disable with FUNC_E2E_CLEAN_IMAGES="false".
+	DefaultCleanImages = true
+
+	// DefaultGocoverdir defines the default path to use for the GOCOVERDIR
+	// while executing tests.  This value can be altered using
+	// FUNC_E2E_GOCOVERDIR. While this value could be passed through using
+	// its original environment variable name "GOCOVERDIR", to keep with the
+	// isolation of environment provided for all other values, this one is
+	// likewise also isolated using the "FUNC_E2E_" prefix.
+	DefaultGocoverdir = "../.coverage"
+
+	// DefaultKubeconfig is the default path (relative to this test file) at
+	// which the kubeconfig can be found which was created when setting up
+	// a local test cluster using the cluster.sh script.  This can be
+	// overridden using FUNC_E2E_KUBECONFIG.
+	DefaultKubeconfig = "../hack/bin/kubeconfig.yaml"
+
+	// DefaultNamespace for E2E tests is that used by default in the
+	// CLI being tested.
+	DefaultNamespace = cmd.DefaultNamespace
```

Comment:
This resolves to "default" aways. It would be nice if this could be configurable (i.e. FUNC_E2E_NAMESPACE) or resolve to current namespace on cluster

Reply:
This is now configurable